Source/DafnyDriver/DafnyNewCli.cs

```diff
@@ -269,40 +268,31 @@ public static class DafnyNewCli {
     }, MiddlewareOrder.Configuration - 101);
     return builder;
   }
-}
 
-class WritersConsole : IConsole {
-  public TextReader InputWriter { get; }
-  public TextWriter ErrWriter { get; }
-  public TextWriter OutWriter { get; }
+  private static async Task<DafnyFile?> HandleDafnyProject(DafnyOptions options,
+    IFileSystem fileSystem, ErrorReporter reporter,
+    Uri uri,
+    IToken origin,
+    bool asLibrary) {
+    if (!asLibrary) {
+      reporter.Error(MessageSource.Project, origin, "Using a Dafny project file as a source file is not supported.");
+      return null;
+    }
 
-  public WritersConsole(TextReader inputWriter, TextWriter outWriter, TextWriter errWriter) {
-    InputWriter = inputWriter;
-    this.ErrWriter = errWriter;
-    this.OutWriter = outWriter;
-  }
+    var outputWriter = new StringWriter();
+    var errorWriter = new StringWriter();
+    var exitCode = await DafnyNewCli.Execute(new WritersConsole(TextReader.Null, outputWriter, errorWriter),
+      new[] { "build", "-t=lib", uri.LocalPath, "--verbose" });
+    if (exitCode != 0) {
+      var output = outputWriter + errorWriter.ToString();
+      reporter.Error(MessageSource.Project, origin,
+        $"Could not build a Dafny library from {uri.LocalPath} because:\n{output}");
+      return null;
+    }
 
-  public IStandardStreamWriter Out => StandardStreamWriter.Create(OutWriter ?? TextWriter.Null);
-
-  public bool IsOutputRedirected => OutWriter != null;
-  public IStandardStreamWriter Error => StandardStreamWriter.Create(ErrWriter ?? TextWriter.Null);
-  public bool IsErrorRedirected => ErrWriter != null;
-  public bool IsInputRedirected => false;
-}
-
-/// <summary>
-/// The class HelpResult is internal to System.CommandLine so we have to include it as source.
-/// It seems System.CommandLine didn't consider having more than one help option as a use-case.
-/// </summary>
-internal class HelpResult : IInvocationResult {
-  public void Apply(InvocationContext context) {
-    var output = context.Console.Out.CreateTextWriter();
-    var helpBuilder = ((HelpBuilder)context.BindingContext.GetService(typeof(HelpBuilder)))!;
-    var helpContext = new HelpContext(helpBuilder,
-      context.ParseResult.CommandResult.Command,
-      output,
-      context.ParseResult);
-
-    helpBuilder.Write(helpContext);
+    var regex = new Regex($"Wrote Dafny library to (.*)\n");
+    var path = regex.Match(outputWriter.ToString());
```

Comment:
Using a regex to parse our own output seems less than ideal. Would it be feasible to add an internal API to run the `build` and read its status/output?